### PR TITLE
Development/static analysis through python

### PR DESCRIPTION
**Pull request by RisingLiberty**
Added a python script that scan for clang-tidy ignored warnings and std usage in the codebase

### CODE REVIEW

File: _build/scripts/static_analysis/static_analysis.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import os
```
Comment (RisingLiberty) [inline _build/scripts/static_analysis/static_analysis.py:1]: This is the only file that needs review

File: _build/scripts/static_analysis/static_analysis.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    regis.diagnostics.log_info(f'scanning for clang-tidy ignore messages')
+    regex_pattern = r'\/\/\s*NOLINT\(' # scanning for "// NOLINT("
+    found_matches : list[FoundMatch] = []
+    for source_root in sourceRoots:
+        found_matches.extend(scan_root(regex_pattern, source_root, excludeFolders))
+
+    ignored_warnings = {}
+
+    for found_match in found_matches:
+        # regis.diagnostics.log_debug(f'{found_match.filepath}({found_match.line_nr}): {found_match.line}')
```
Comment (Dyronix) [inline _build/scripts/static_analysis/static_analysis.py:54]: Please don't push commented code

File: _build/scripts/static_analysis/static_analysis.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        regis.diagnostics.log_info(f'{found_match.filepath}({found_match.line_nr}): {found_match.line.rstrip()}')
+
+def log_header(msg : str):
+    regis.diagnostics.log_info('------------------------')
+    regis.diagnostics.log_info(msg)
+    regis.diagnostics.log_info('------------------------')
+
+def main():
+    log_header(f'scanning for clang-tidy ignores')
+    scan_clang_tidy_ignore(source_root_folders, exclude_folders)
```
Comment (Dyronix) [inline _build/scripts/static_analysis/static_analysis.py:87]: INFO: Why do we need this is this just for logging purposes that we know what is getting ignored?

**Review by Dyronix — CHANGES_REQUESTED** [review]

File: _build/scripts/static_analysis/static_analysis.py (hunk 3 above)
Comment (RisingLiberty) [reply _build/scripts/static_analysis/static_analysis.py:87]: I plan on adding it to CI in a way, maybe automatically submit it into version control so we can see what we ignore and where etc, if we ever decide to change our clang-tidy flags, we can base it on that data

**Review by Dyronix — APPROVED** [review]

**Pull request merged**